libfwupd/fwupd-client.c

```diff
@@ -5380,6 +5421,36 @@ fwupd_client_download_http(FwupdClient *self, CURL *curl, const gchar *url, GErr
 	return g_bytes_new(buf->data, buf->len);
 }
 
+static gboolean
+fwupd_client_download_error_is_fatal(const GError *error)
+{
+	if (g_error_matches(error, FWUPD_ERROR, FWUPD_ERROR_TIMED_OUT))
+		return FALSE;
+	return TRUE;
+}
+
+static GBytes *
+fwupd_client_download_http_retry(FwupdClient *self, CURL *curl, const gchar *url, GError **error)
+{
+	FwupdClientPrivate *priv = GET_PRIVATE(self);
+	guint delay_ms = 2500;
+	for (guint i = 0;; i++, delay_ms *= 2) {
```

Comment:
why increase the delay every time?

Reply:
in case the user does something like `fwupdmgr download http://foo.com --download-retries 1000000`